crates/core/tedge_agent/src/operation_workflows/persist.rs

```diff
@@ -289,6 +297,22 @@ impl WorkflowRepository {
         }
     }
 
+    async fn release_in_use_copy(&mut self, operation: &OperationName, version: &WorkflowVersion) {
+        if let Some(count) = self.in_use_copies.get_mut(version) {
+            *count -= 1;
+            if *count == 0 {
```

Comment:
```suggestion
            if *count != 0 {
```
We should proceed with the removal only when the count become `0`, right?